analyzers/SonarAnalyzer.Shared/Extensions/SyntaxTokenExtensions.cs

```diff
@@ -0,0 +1,88 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2021 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using Microsoft.CodeAnalysis;
+
+#if CS
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+#else
+using Microsoft.CodeAnalysis.VisualBasic.Syntax;
+#endif
+
+namespace SonarAnalyzer.Extensions
+{
+    internal static class SyntaxTokenExtensions
+    {
+        // Based on Roslyn: https://github.com/dotnet/roslyn/blob/09903da31892a30f71eab67d2fd83232cfbf0cea/src/Workspaces/CSharp/Portable/LanguageServices/CSharpSyntaxFactsService.cs#L1187-L1254
+        public static SyntaxNode GetBindableParent(this SyntaxToken token)
+        {
+            var node = token.Parent;
+
+            while (node != null)
+            {
+                var parent = node.Parent;
+
+                // If this node is on the left side of a member access expression, don't ascend
+                // further or we'll end up binding to something else.
+                if (parent is MemberAccessExpressionSyntax memberAccess &&
+                    memberAccess.Expression == node)
```

Comment:
To fit our coding style (&& should be at the beginnig of the line). Conditions are short so let's make them single line.
```suggestion
                if (parent is MemberAccessExpressionSyntax memberAccess &&                    memberAccess.Expression == node)
```

Reply:
The first 4 looks like a candidate for some switch expression, but I didn't tried that so I don't know how that would play with the rest.

Reply:
The problem is the branch at line 81 which does not return. It updates the parent to re-enter the loop.

I've reorganized the code a bit with 2 switch statements.

Reply:
Nice :shipit: